daemon/cmd/map.go

```diff
@@ -5,13 +5,71 @@ package cmd
 
 import (
 	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/strfmt"
 
 	"github.com/cilium/cilium/api/v1/models"
 	restapi "github.com/cilium/cilium/api/v1/server/restapi/daemon"
 	"github.com/cilium/cilium/pkg/bpf"
 	"github.com/cilium/cilium/pkg/ebpf"
 )
 
+type eventsDumper interface {
+	DumpEventsWithCallback(bpf.EventCallbackFunc) error
+}
+
+type mapRefGetter interface {
+	GetMap(name string) (eventsDumper, bool)
+}
+
+type mapGetterImpl struct{}
+
+func (mg mapGetterImpl) GetMap(name string) (eventsDumper, bool) {
+	m := bpf.GetMap(name)
+	return m, m != nil
+}
+
+type getMapNameEvents struct {
+	daemon    *Daemon
+	mapGetter mapRefGetter
+}
+
+func NewGetMapNameEventsHandler(d *Daemon, maps mapRefGetter) restapi.GetMapNameEventsHandler {
+	return &getMapNameEvents{
+		daemon:    d,
+		mapGetter: maps,
+	}
+}
+
+func (h *getMapNameEvents) Handle(params restapi.GetMapNameEventsParams) middleware.Responder {
+	m, exists := h.mapGetter.GetMap(params.Name)
+	if !exists {
+		return restapi.NewGetMapNameNotFound()
+	}
+	mapEvents := []*models.MapEvent{}
+	err := m.DumpEventsWithCallback(func(e *bpf.Event) {
```

Comment:
Do you have to worry about any racing with goroutines updating the map here?

Reply:
DumpEventsWithCallback holds the bpf.Maps readlock during the callback.